### ISSUE TITLE
[V2] Passing props to custom top bar component doesn't work as expected

### ISSUE THREAD

**Issue by jarrx** [wix/react-native-navigation#3622]
### Issue Description

Sometimes the passed prop is undefined after reloading the app. Restarting the app usually makes it work again. This happens from time to time.
Also, what I'm trying to do is passing a reference to an animated value to make a part of my custom top bar component animate when scrolling. Usually there's no problem passing a reference directly to a child component, but when passing to the top bar component it doesn't work. Even if the prop is not undefined, the animated value doesn't update.
This could be solved by passing an function that returns the animated value instead, but is this expected behavior?

### Steps to Reproduce / Code Snippets / Screenshots

```
  Navigation.mergeOptions(this.props.componentId, {
      topBar: {
        component: {
          name: 'topBar',
          passProps: { animatedScrollPositionX: this.animatedScrollPositionX },
        }
      }
    });
```

---
### Environment
* React Native Navigation version: ^2.0.2394
* React Native version: ^0.55.4
* Platform(s) (iOS, Android, or both?): IOS, android?
* Device info (Simulator/Device? OS version? Debug/Release?): Simulator

**Comment by reganperkins** [comment]
where you able to successfully you able to use a custom topbar in android?

**Comment by jarrx** [comment]
@reganperkins No, I haven't tried on android. (sorry for my misleading environment description).
